src/Lean/Elab/Tactic/BuiltinTactic.lean

```diff
@@ -369,12 +369,36 @@ private def getCaseGoals (tag : TSyntax ``binderIdent) : TacticM (MVarId × List
   let gs ← getUnsolvedGoals
   let g ← if let `(binderIdent| $tag:ident) := tag then
     let tag := tag.getId
-    let some g ← findTag? gs tag | throwError "tag not found"
+    let some g ← findTag? gs tag | notFound tag gs
     pure g
   else
     getMainGoal
   return (g, gs.erase g)
 
+  where
+    -- When the case tag is not found, construct a message that tells
+    -- the user what they could have written
+    notFound (tag : Name) (available : List MVarId) := do
+      let firstLine := m!"Case tag {tag} not found."
```

Comment:
I believe we usually put names like `{tag}` from the input in single quotes in messages. Not sure how consistent we are about it though, so I'll leave making the call on it to you.

Reply:
I would certainly prefer this - I'm updating the code now.